woocommerce/woocommerce
# [Experimental] Allow overriding placeholder in interactivity Dropdown::render

### Changes proposed in this Pull Request:

In my [previous PR](https://github.com/woocommerce/woocommerce/pull/43183#discussion_r1438998009) @dinhtungdu mentioned it would be good to be able to override the placeholder text provided to the Dropdown.php interactivity component.

This provides that functionality. We now determine the default placeholder text once from in PHP instead of front-end (and pass it as context to front-end) and if you want to override it you can pass it as a prop of `placeholder`.

This also adjusts the attribute filter to work with it and fixes a couple of bugs I noticed in the filter when I was working with it in Dropdown mode.

Also note there is a behaviour I'm not sure should be correct which is that the attributes available to choose from in the dropdown get filtered down to those available in the currently filtered products. I'm not sure that's desirable behavior for this kind of filter. wdyt @dinhtungdu ?

Closes #43205



### How to test the changes in this Pull Request:


1. Create a collection filters block
2. Select the attribute filter and in the tools, choose "dropdown" for the mode.
3. Preview in front-end and see that the placeholder matches the attribute name, and is kept up to date if you select then deselect an attribute from the dropdown.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
[Experimental] support passing a placeholder text to the interactivity Dropdown component.

#### Comment 

</details>
